admin/src/pages/HomePage/index.js

```diff
@@ -14,33 +14,61 @@ import { Badge } from "@strapi/design-system/Badge";
 import { Typography } from "@strapi/design-system/Typography";
 import { Link } from "@strapi/design-system/Link";
 import { useIntl } from "react-intl";
-import { getTrad } from "../../utils";
-import getBlocks from "../../api/getBlocks";
+import { getTrad, axiosInstance } from "../../utils";
 
 const ImageWrapper = styled.div`
     cursor: pointer;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: 100%;
 `;
 
+const Image = styled.image`
+    height: 250px;
+    width: 100%;
+    background-image: url(${(props) => props.src});
+    background-size: contain;
+    background-repeat: no-repeat;
+    background-position: center;
+`;
 const HomePage = () => {
     const [isLoading, setIsLoading] = useState(true);
-    const [blocks, setBlocks] = useState([]);
+    const [blocks, setBlocks] = useState(null);
+    const [strapiBlocks, setStrapiBlocks] = useState(null);
+    const [databaseBlocks, setDatabaseBlocks] = useState(null);
     const [activeId, setActiveId] = useState(0);
     const [modalOpen, setModalOpen] = useState(false);
     const { formatMessage } = useIntl();
 
-    useEffect(async () => {
-        const blocksArr = await getBlocks();
-        if (blocksArr.length > 0) {
-            setBlocks(blocksArr);
-        }
-        if (blocksArr) {
-            setIsLoading(false);
+    useEffect(() => {
+        const data = axiosInstance.get(`/block-gallery/blocks`);
+        if (data) {
+            data.then((res) => {
+                setStrapiBlocks(res.data.strapiBlocks);
+                setDatabaseBlocks(res.data.databaseBlocks);
+                setBlocks(res.data.blocks);
+            });
+            if (blocks && strapiBlocks === databaseBlocks) {
+                setIsLoading(false);
+            } else {
+                setIsLoading(true);
+                const secondaryData = axiosInstance.get(
+                    `/block-gallery/blocks`
+                );
+                secondaryData.then((res) => {
+                    setStrapiBlocks(res.data.strapiBlocks);
+                    setDatabaseBlocks(res.data.databaseBlocks);
+                    setBlocks(res.data.blocks);
+                });
+                setIsLoading(false);
+            }
         }
```

Comment:
@jbednar97 toto predsa ale tiez nie je dobre. Jedna vec, preco nedas then rovno k tomu axiosInstance, ale davas to este do data ? Ved predsa if (data) bude vzdy true, to bude vzdy promise. Vsetky 3 premenne, ktore kontrolujes na riadku 52 nemusia mat spravne hodnoty, ale tie predchadzajuce, pretoze tesne predtym volas settery, preco to neni cele v tej then casti a nepracujes s datami, ktore su v tom res ? A aj to by som urobil teda .then(({ data})) => {}. Hlavne jako preco to volas 2x ? Jaky presne ma zmysel to, ze 2x nacitavas data z endpointu /block-gallery/blocks ?